tests/test_helpers.py

```diff
@@ -81,3 +96,171 @@ def test_get_font_stack_add_emoji_true(font_stack_names):
     assert all(
         extended_emoji.issubset(_get_font_stack(name, add_emoji=True)) for name in font_stack_names
     )
+
+
+@pytest.mark.parametrize(
+    "units, x_out",
+    [
+        ("x10^7 m s^-1", ["x10^7", "m", "s^-1"]),
+        (
+            "-20 kg^2 m_12 m[_0^2] g/L %C6H12O6% x10^-3",
+            ["-20", "kg^2", "m_12", "m[_0^2]", "g/L", "%C6H12O6%", "x10^-3"],
+        ),
+        (
+            "  -20  _kg_^2      *m*_12  m[_0^2] g/L%C6H12O6% x10^-3",
+            ["-20", "_kg_^2", "*m*_12", "m[_0^2]", "g/L%C6H12O6%", "x10^-3"],
+        ),
+        (
+            "m^-3 :cdot: kg :cdot: m[_0^2] *per:space:mille*",
+            ["m^-3", ":cdot:", "kg", ":cdot:", "m[_0^2]", "*per:space:mille*"],
```

Comment:
I don't understand what different things are being tested across these cases. The 3rd seems to be testing tokens splitting multiple spaces, but that doesn't seem to be all it tests. Please break the tests into...

* tests of small and specific bits of logic
* new tests should test / clarify new pieces of behavior

It's okay to have some bigger inputs tested, but it should be clear / quickly explainable what specifically is being tested / what's new compared to prev tests.

Reply:
This is now much simpler and much improved!